src/utils.ts

```diff
@@ -1,41 +1,39 @@
-export type Bytes = string;
+export type Bytes = Uint8Array | string;
 
-export function u8ArrayToBytes(array: Uint8Array) {
-  let ret = "";
-  for (let e of array) {
-    ret += String.fromCharCode(e);
-  }
-  return ret;
-}
-
-// TODO this function is a bit broken and the type can't be string
-// TODO for more info: https://github.com/near/near-sdk-js/issues/78
 export function bytesToU8Array(bytes: Bytes): Uint8Array {
-  let ret = new Uint8Array(bytes.length);
-  for (let i = 0; i < bytes.length; i++) {
-    ret[i] = bytes.charCodeAt(i);
+  if (typeof bytes === 'string') {
+    return latin1ToU8Array(bytes);
   }
-  return ret;
+  return bytes;
 }
 
-export function bytes(strOrU8Array: string | Uint8Array): Bytes {
-  if (typeof strOrU8Array == "string") {
-    return checkStringIsBytes(strOrU8Array);
-  } else if (strOrU8Array instanceof Uint8Array) {
-    return u8ArrayToBytes(strOrU8Array);
+export function u8ArrayToLatin1(array: Uint8Array): string {
+  let ret = "";
+  for (let e of array) {
+    ret += String.fromCharCode(e);
   }
-  throw new Error("bytes: expected string or Uint8Array");
+  return ret;
```

Comment:
```suggestion
  return String.fromCharCode(...array);
```

And you can delete the lines above